server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -571,8 +571,10 @@ private static void applyStartedShards(RoutingAllocation routingAllocation, List
                     + startedShard
                     + " but was: "
                     + routingNodes.getByAllocationId(startedShard.shardId(), startedShard.allocationId().getId());
-
-            routingNodes.startShard(logger, startedShard, routingAllocation.changes());
+            long expectedShardSize = routingAllocation.metadata().getIndexSafe(startedShard.index()).isSearchableSnapshot()
```

Comment:
I think this is always `ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE`? I cannot find where we populate this with the value from `org.elasticsearch.snapshots.SnapshotShardSizeInfo#getShardSize(org.elasticsearch.cluster.routing.ShardRouting, long)`. Am I missing something?

Reply:
It gets set in `ShardRouting#moveToStarted`

Reply:
I see it know, thanks for clarifying!